### PR TITLE
Support R graphics engine 16

### PR DESCRIPTION
**Pull request by timtmok**
### Intent
Address #12751 

### Approach
Changes made according to https://github.com/rstudio/rstudio-pro/wiki/RStudio-Graphics-Device#preparing-for-an-r-update

### Automated Tests
none

### QA Notes
Rendering a plot will show in the Plots pane. Loading a library that uses the graphics engine like ggplot2 should not show a warning about the graphics engine version.

### Documentation
Updated README

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/r/session/graphics/RGraphicsDevDescVersions.hpp [hunk 1]
```diff
@@ -777,6 +777,131 @@ struct DevDescVersion15
...
+    void (*useGroup)(SEXP ref, SEXP trans, pDevDesc dd);
+    void (*releaseGroup)(SEXP ref, pDevDesc dd);
+
+    void (*stroke)(SEXP path, const pGEcontext gc, pDevDesc dd);
+    void (*fill)(SEXP path, int rule, const pGEcontext gc, pDevDesc dd);
+    void (*fillStroke)(SEXP path, int rule, const pGEcontext gc, pDevDesc dd);
+
+    SEXP (*capabilities)(SEXP cap);
+
+    void (*glyph)(int n, int *glyphs, double *x, double *y,
```
Comment (kevinushey) [inline src/cpp/r/session/graphics/RGraphicsDevDescVersions.hpp:899]:
```suggestion
    // below added in version 16 (R 4.3.0)
    void (*glyph)(int n, int *glyphs, double *x, double *y, 
```

File: src/cpp/r/session/graphics/RGraphicsDevDesc.cpp [hunk 2]
```diff
@@ -264,11 +264,47 @@ pDevDesc allocate(const RSDevDesc& devDesc)
...
+      pDD->releaseMask = devDesc.releaseMask;
+
+      pDD->defineGroup = devDesc.defineGroup;
+      pDD->useGroup = devDesc.useGroup;
+      pDD->releaseGroup = devDesc.releaseGroup;
+      pDD->stroke = devDesc.stroke;
+      pDD->fill = devDesc.fill;
+      pDD->fillStroke = devDesc.fillStroke;
+      pDD->capabilities = devDesc.capabilities;
+
```
Comment (kevinushey) [inline src/cpp/r/session/graphics/RGraphicsDevDesc.cpp:300]:
```suggestion

pDD->glyph = devDesc.glyph;
```

Comment (timtmok) [reply src/cpp/r/session/graphics/RGraphicsDevDesc.cpp:300]: I woke up and realized I think I forgot something!

**Review by melissa-barca — APPROVED** [review]
LGTM!

**Pull request merged**